server/src/main/java/org/elasticsearch/common/bytes/CompositeBytesReference.java

```diff
@@ -187,6 +188,13 @@ public BytesRef next() throws IOException {
         };
     }
 
+    @Override
+    public void writeTo(OutputStream os) throws IOException {
+        for (BytesReference reference : references) {
```

Comment:
Taking advantage of the now cheaper `writeTo` for the `BytesArray` + this is always cheaper since we never cleverly batch up writes from multiple components anyway